Jasmine/Common/Models/Coordinate.swift

```diff
@@ -12,17 +12,40 @@ struct Coordinate {
 
     /// Default constructor for Coordinate.
     init(row: Int, col: Int) {
-        guard row >= 0, col >= 0 else {
-            fatalError("Row and column coordinates should be >= 0")
-        }
         self.row = row
         self.col = col
     }
 
-    /// Returns the Coordinate of the next row
+    /// Returns the Coordinate of the next row, which is one row down.
     var nextRow: Coordinate {
         return Coordinate(row: row + 1, col: col)
     }
+
+    /// Returns the Coordinate of the previous row, which is one row up.
+    var prevRow: Coordinate {
+        return Coordinate(row: row - 1, col: col)
+    }
+
+    /// Returns the Coordinate of the next column, which is one column to the right.
+    var nextCol: Coordinate {
+        return Coordinate(row: row, col: col + 1)
+    }
+
+    /// Returns the Coordinate of the previous column, which is one column to the left.
+    var prevCol: Coordinate {
+        return Coordinate(row: row, col: col - 1)
+    }
+
+    /// Gets the index path from the current coordinate.
+    var toIndexPath: IndexPath {
+        return IndexPath(item: col, section: row)
```

Comment:
And `Coordinate` is pretty much used everywhere, not only model specific actually.